docs/shared/recipes/plugins/project-graph-plugins.md

```diff
@@ -21,113 +32,215 @@ You can register a plugin by adding it to the plugins array in `nx.json`:
-  - `filesToProcess` contains all the files that have changed since the last invocation and need to be reanalyzed.
+### Identifying Projects
 
-The `processProjectGraph` function should return an updated `ProjectGraph`. This is most easily done using `ProjectGraphBuilder`. The builder is there for convenience, so you don't have to use it.
+Looking at the tuple, you can see that the first element is a file pattern. This is a glob pattern that Nx will use to find files in your workspace. The second element is a function that will be called for each file that matches the pattern. The function will be called with the path to the file and a context object. Your plugin can then return a set of projects and external nodes.
 
-```typescript
-import {
-  ProjectGraph,
-  ProjectGraphBuilder,
-  ProjectGraphProcessorContext,
-  DependencyType,
-} from '@nx/devkit';
-
-export function processProjectGraph(
-  graph: ProjectGraph,
-  context: ProjectGraphProcessorContext
-): ProjectGraph {
-  const builder = new ProjectGraphBuilder(graph);
-  // We will see how this is used below.
-  return builder.getUpdatedProjectGraph();
-}
-```
+If a plugin identifies a project that is already in the project graph, it will be merged with the information that is already present. The builtin plugins that identify projects from `package.json` files and `project.json` files are ran after any plugins listed in `nx.json`, and as such will overwrite any configuration that was identified by them. In practice, this means that if a project has both a `project.json`, and a file that your plugin identified, the settings the plugin idenfied will be overwritten by the `project.json`'s contents.
 
-## Adding New Nodes to the Project Graph
+Project nodes in the graph are considered to be the same if the project has the same root. If multiple plugins identify a project with the same root, the project will be merged. In doing so, the name that is already present in the graph is kept, and the properties below are shallowly merged. Any other properties are overwritten.
 
-You can add nodes to the project graph. Since first-party code is added to the graph automatically, this is most commonly used for third-party packages.
+- `targets`
+- `tags`
+- `implicitDependencies`
+- `generators`
 
-A Project Graph Plugin can add them to the project graph. After these packages are added as nodes to the project graph, dependencies can then be drawn from the workspace projects to the third party packages as well as between the third party packages.
+Note: This is a shallow merge, so if you have a target with the same name in both plugins, the target from the second plugin will overwrite the target from the first plugin. Options, configurations, or any other properties within the target will be overwritten **_not_** merged.
+
+#### Example
+
+A simplified version of the project.json plugin is shown below, which adds a new project to the project graph for each `project.json` file it finds.
```

Comment:
What are they supposed to do with the `createNodes` variable?  How do they register it with Nx?